src/http/router.zig

```diff
@@ -1,362 +1,53 @@
-                .{std.mem.trimRight(u8, url_path, "/")},
-            );
-
-            try self.serve_route(url_with_match_all, route);
-        }
-
-        pub fn serve_embedded_file(
-            self: *Self,
-            comptime path: []const u8,
-            comptime mime: ?Mime,
-            comptime bytes: []const u8,
-        ) !void {
-            assert(!self.locked);
-            const route = Route.init().get({}, struct {
-                fn handler_fn(ctx: *Context, _: void) !void {
-                    const cache_control: []const u8 = if (comptime builtin.mode == .Debug)
-                        "no-cache"
-                    else
-                        comptime std.fmt.comptimePrint(
-                            "max-age={d}",
-                            .{std.time.s_per_day * 30},
-                        );
-
-                    ctx.response.headers.putAssumeCapacity("Cache-Control", cache_control);
-
-                    // If our static item is greater than 1KB,
-                    // it might be more beneficial to using caching.
-                    if (comptime bytes.len > 1024) {
-                        @setEvalBranchQuota(1_000_000);
-                        const etag = comptime std.fmt.comptimePrint("\"{d}\"", .{std.hash.Wyhash.hash(0, bytes)});
-                        ctx.response.headers.putAssumeCapacity("ETag", etag[0..]);
-
-                        if (ctx.request.headers.get("If-None-Match")) |match| {
-                            if (std.mem.eql(u8, etag, match)) {
-                                try ctx.respond(.{
-                                    .status = .@"Not Modified",
-                                    .mime = Mime.HTML,
-                                    .body = "",
-                                });
-
-                                return;
-                            }
-                        }
-                    }
-
-                    try ctx.respond(.{
-                        .status = .OK,
-                        .mime = mime,
-                        .body = bytes,
-                    });
-                }
-            }.handler_fn);
-
-            try self.serve_route(path, route);
+            return self;
         }
 
         pub fn serve_not_found(self: *Self, route: Route) void {
             assert(!self.locked);
             self.not_found_route = route;
         }
 
-        pub fn serve_route(self: *Self, path: []const u8, route: Route) !void {
-            assert(!self.locked);
-            try self.routes.add_route(path, route);
-        }
-
         pub fn get_route_from_host(self: Self, path: []const u8, captures: []Capture, queries: *QueryMap) !FoundRoute {
-            const base_404_route = comptime Route.init().get({}, struct {
-                fn not_found_handler(ctx: *Context, _: void) !void {
+            const base_404_route = comptime Route.init("").get(struct {
+                fn not_found_handler(ctx: *Context) !void {
```

Comment:
Now that routes are comptime-known, we could do this at comptime too. However, as it's an optional parameter, I see 2 ways to define a custom not found handler at comptime:
1. Add a configuration structure passed in init, that could be empty if the user doesn't wish to set a custom not found handler.
2. Change the router API to make a "router builder" which will allow to call `.withNotFound` to define a custom handler for not found handler, before `init`.

```zig

// Option 1
var router = Router.init(&[_]Route{...}, .{
    .not_found = notFoundHandler,
});

// Option 2
var router = Router.withNotFound(notFoundHandler).init(&[_]Route{...});


```

Reply:
I tried to implement the second option, to see how it would look like, you can see it here: https://github.com/Madeorsk/zzz/compare/comptime-router...comptime-router-builder#diff-25b613b3e4187e4947ff5bec65f4435bde9da62a22ac53cbf246ea681dd92e78

Reply:
I'm thinking option 1 for now. At some point, routes are going to get another pass to have layering working (especially with middleware) and that'll will be the point where a `RouterBuilder` makes sense.

The scope of this PR should be the comptime functionality and then that route layering (with the builder) should be a future one.

Reply:
OK :-) I'm keeping my builder implementation aside for the middlewares (that I will make soon, but there are a few other small PR that I need to push first, and I'll do it one by one so you can spot all the changes more easily ; this one is probably the biggest one...).